ethers-contract/ethers-contract-abigen/src/contract.rs

```diff
@@ -151,6 +154,17 @@ impl Context {
             }
         }
 
+        let mut event_aliases = BTreeMap::new();
+        for (signature, alias) in args.event_aliases.into_iter() {
+            let alias = syn::parse_str(&alias)?;
+            if event_aliases.insert(signature.clone(), alias).is_some() {
+                return Err(anyhow!(
+                    "duplicate method signature '{}' in method aliases",
+                    signature,
+                ));
+            }
```

Comment:
This is redundant for `event_aliases` since it is a HashMap, and `signature` is already unique.
Probably best to refactor like
```suggestion
            event_aliases.insert(signature.clone(), alias);
```